### PR TITLE
feat(api): support for configuring several filetypes in one go

### LINKED ISSUE #14: accept array for filetype
```lua
ft('javascript'):fmt('prettier')
ft('typescript'):fmt('prettier')
ft('javascriptreact'):fmt('prettier')
ft('typescriptreact'):fmt('prettier')
```

->

```lua
ft({ 'javascript', 'typescript', 'javascriptreact', 'typescriptreact' }):fmt('prettier')
```

### PR DESCRIPTION
**Pull request by xiaoshihou514**
Resolves #14 
But it basically doubles the code in filetype.lua, so I'm not sure if it should be merged, if it does I will update the README.

### CODE REVIEW

File: lua/guard/filetype.lua [hunk 1]
```diff
@@ -35,8 +35,49 @@ local function box()
   return setmetatable({}, tbl)
 end
 
+
+local function box_for_group(fts)
+  for _, ft in pairs(fts) do
```
Comment (glepnir) [inline lua/guard/filetype.lua:40]: for list-like table use `ipairs` and could you write a test?

File: lua/guard/filetype.lua [hunk 2]
```diff
@@ -37,7 +37,7 @@ end
 
 
 local function box_for_group(fts)
```
Comment (glepnir) [inline lua/guard/filetype.lua:39]: I am very confuse on there. can we reuse exist function instead of create `box_for_group` lots of things are duplicated

Comment (xiaoshihou514) [reply lua/guard/filetype.lua:39]: Well yes and no.
The object returned by `ft({"foo", "bar"})` _would_ have to use a different set of logic. But the fmt, lint and append methods are really just calling the respective methods on `ft("foo")` and `ft("bar")` (in this case). I don't know if the logic could be simplified further.

Comment (xiaoshihou514) [reply lua/guard/filetype.lua:39]: Maybe the metatable related logic could be abstracted tho.

**Pull request closed without merging**